contracts/cw20-atomic-swap/src/contract.rs

```diff
@@ -66,13 +66,14 @@ pub fn try_create<S: Storage, A: Api, Q: Querier>(
     deps: &mut Extern<S, A, Q>,
     env: Env,
     msg: CreateMsg,
-    balance: Balance,
+    mut balance: Balance,
 ) -> StdResult<HandleResponse> {
     if !is_valid_name(&msg.id) {
         return Err(StdError::generic_err("Invalid atomic swap id"));
     }
 
-    // FIXME: normalize array first (remove zero-valued coins), and then check for empty
+    // Normalize coins first
+    balance.normalize();
```

Comment:
Maybe we should avoid the two calls and just make balance.is_empty more intelligent. That would make it hard to anyone else using this to get it wrong.

Reply:
I thought about that, but they are different operations. And, they are being called separately in the subkeys contract.

Anyway, I don't think we'll never call `is_empty()` more than once per call / endpoint. So, I'm OK with joining them.

Reply:
With the change to is_empty below, you can remove this line, right?